sdk/tables/azure-data-tables/tests/test_table_client.py

```diff
@@ -253,8 +252,76 @@ def test_client_with_sas_token(self, tables_storage_account_name, tables_primary
             )
             for e in entities:
                 pass
+
+        with pytest.raises(ValueError) as ex:
+            client = TableClient.from_table_url(sas_url, credential=AzureSasCredential(sas_token))
+        ex_msg = "You cannot use AzureSasCredential when the resource URI also contains a Shared Access Signature."
+        assert ex_msg == str(ex.value)
+
+        with TableClient.from_table_url(sas_url, credential=self.get_token_credential()) as client:
+            entities = client.query_entities(
+                query_filter="PartitionKey eq @pk",
+                parameters={"pk": "dummy-pk"},
+            )
+            for e in entities:
+                pass
+
+        with TableClient.from_table_url(sas_url, credential=tables_primary_storage_account_key) as client:
```

Comment:
Same question, which credential is actually being used to authenticate here?

Reply:
AzureNamedKeyCredential